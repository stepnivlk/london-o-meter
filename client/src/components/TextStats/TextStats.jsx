// @flow

import React from 'react';

import type { Item } from '../../App';

type Props = {
  data: Item[],
  highlightedIndex: number,
};

const styleForStat = (item, isHighlighted) => ({
  color: item.color,
  fontSize: isHighlighted ? 30 : 20,
  fontWeight: isHighlighted ? 'bold' : 'normal'
});

const Stat = ({ item, text, isHighlighted }) => (
  <span style={styleForStat(item, isHighlighted)}>
    {item.valuePercent}% {text}
  </span>
);

const findItem = (label, data) => data.find(item => item.label === label);

const nullItem = {
  index: null,
  color: '',
  valuePercent: null,
}

const TextStats = ({ data, highlightedIndex }: Props) => {
  const happyItem = findItem('positive', data) || nullItem;
  const unhappyItem = findItem('negative', data) || nullItem;

  return (
    <div>
      <Stat
        item={unhappyItem}
        text='Unhappy'
        isHighlighted={unhappyItem.index === highlightedIndex}
      />
      {' / '}
      <Stat
        item={happyItem}
        text='Happy'
        isHighlighted={happyItem.index === highlightedIndex}
      />
    </div>
  )
}

export default TextStats;
