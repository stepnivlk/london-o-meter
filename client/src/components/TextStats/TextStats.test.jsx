import React from 'react';
import renderer from 'react-test-renderer';
import TextStats from './TextStats';

const createItem = (label, index) => ({
  label,
  value: 10,
  valuePercent: 50,
  index,
  color: '#fff',
  isActive: false
});

it('matches a snapshot', () => {
  const component = renderer.create(
    <TextStats
      data={[createItem('positive', 0), createItem('negative', 1)]}
      highlightedIndex={0}
    />
  );

  const tree = component.toJSON();

  expect(tree).toMatchSnapshot();
});

it('Matches expected style', () => {
  const component = renderer.create(
    <TextStats
      data={[createItem('positive', 0), createItem('negative', 1)]}
      highlightedIndex={0}
    />
  );

  const tree = component.toJSON();

  const firstStatStyle = tree.children[0].props.style;
  const secondStatStyle = tree.children[2].props.style;

  const expectedActiveStyle = { color: '#fff', fontSize: 30, fontWeight: 'bold' };
  const expectedNonactiveStyle = { color: '#fff', fontSize: 20, fontWeight: 'normal' };

  expect(firstStatStyle).toMatchObject(expectedNonactiveStyle);
  expect(secondStatStyle).toMatchObject(expectedActiveStyle);
});
