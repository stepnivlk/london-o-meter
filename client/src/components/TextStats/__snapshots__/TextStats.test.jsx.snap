// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches a snapshot 1`] = `
<div>
  <span
    style={
      Object {
        "color": "#fff",
        "fontSize": 20,
        "fontWeight": "normal",
      }
    }
  >
    50
    % 
    Unhappy
  </span>
   / 
  <span
    style={
      Object {
        "color": "#fff",
        "fontSize": 30,
        "fontWeight": "bold",
      }
    }
  >
    50
    % 
    Happy
  </span>
</div>
`;
